### PR TITLE
chore: group dependabot PRs

### LINKED ISSUE #878: group dependabot prs
**Describe the solution you'd like**
https://github.com/eraser-dev/eraser/blob/main/.github/dependabot.yml

group where it makes sense

for example:
- gh actions group
- npm group

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Eraser version:
- Kubernetes version: (use `kubectl version`):

### PR DESCRIPTION
**Pull request by ashnamehrotra**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #878

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -26,6 +34,10 @@ updates:
         update-types:
         - "version-update:semver-major"
         - "version-update:semver-minor"
+    groups:
+      all:
+        patterns:
+        - "*"
```
Comment (sozercan) [inline .github/dependabot.yml]: we shouldn't do * here, but only for things we want to group (for example k8s.io) but I remember CRI had to be specific version to not break stuff

File: .github/dependabot.yml [hunk 2]
```diff
@@ -7,13 +7,21 @@ updates:
       interval: "weekly"
     commit-message:
       prefix: "chore"
+    groups:
+      all:
+        patterns:
+        - "*"
```
Comment (sozercan) [inline .github/dependabot.yml]: might want to group specific things here too, like docuausuarus ones instead of *

File: .github/dependabot.yml [hunk 3]
```diff
@@ -26,6 +34,10 @@ updates:
         update-types:
         - "version-update:semver-major"
         - "version-update:semver-minor"
+    groups:
+      k8s:
+        patterns:
+        - "k8s.io/kubernetes"
```
Comment (sozercan) [inline .github/dependabot.yml]: this is a single dependency. we can do k8s.io/* but we'll have to exclude cri. looks like we can have exclusions: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file

Comment (ashnamehrotra) [reply .github/dependabot.yml]: ah got it! will change it to exclude cri

**Review by sozercan — APPROVED** [review]
Thanks! LGTM

**Pull request merged**